src/common/guac_surface.c

```diff
@@ -435,8 +447,11 @@ static int __guac_common_surface_should_use_webp(guac_common_surface* surface,
     if (!guac_client_supports_webp(surface->client))
         return 0;
 
-    /* Usage criteria are currently the same as JPEG */
-    return __guac_common_surface_should_use_jpeg(surface, rect);
+    /* Calculate the average framerate for the given rect */
+    int framerate = __guac_common_surface_calculate_framerate(surface, rect);
+
+    /* WebP is preferred if framerate is high enough */
+    return framerate >= GUAC_COMMON_SURFACE_JPEG_FRAMERATE;
```

Comment:
Ok, will do.

Your last comment caught my attention though; when you scroll, lossy (adaptive)encoding kicks in. 
- In the JPEG code path; for a terminal window the png_optimality code will then prefer PNG. It will compress better (and look better as a side-effect).
- In the WebP code path; lossy WebP should compress better than PNG. The nasty side-effect of lossy WebP is that the terminal window looks bad.

The way we solved this is through 2 refresh pipes; by repainting areas of the screen which were painted lossy with a non-lossy refresh for those areas when the frequent refreshes stops.